source/parser.hera

```diff
@@ -6649,20 +6649,31 @@ Init
     module.wrapIIFE = (exp, async) => {
       let prefix, suffix
       if (async) {
-        prefix = "(async ()=>{"
-        suffix = "})()"
+        prefix = "(async ()=>"
+        suffix = ")()"
       } else if (hasAwait(exp)) {
-        prefix = "(await (async ()=>{"
-        suffix = "})())"
+        prefix = "(await (async ()=>"
+        suffix = ")())"
       } else {
-        prefix = "(()=>{"
-        suffix = "})()"
+        prefix = "(()=>"
+        suffix = ")()"
       }
-      if (Array.isArray(exp)) {
-        return [prefix, ...exp, suffix]
-      } else {
-        return [prefix, exp, suffix]
+
+      // TODO: This rest prevents an infinite recursion bug, ideally it wouldn't be necessary
+      const expressions = Array.isArray(exp) ? [[...exp]] : [exp]
```

Comment:
Could we get away with `[...exp]` in the first case?

I admit I don't yet see where infinite recursion would arise, but trust that aliasing is a tough problem...

Reply:
`[...exp]` caused a problem of expanding `[ws, exp, delimiter]` statement tuples that were passed, causing a problem with inserting the hoisted ref. Unifying the statement nodes should help with that.

Now that we're using `lib.js` we could even update to `lib.ts` or `lib.civet` someday.